tasks/eva_2155/correct_contig_error_in_study.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import hashlib
+import logging
+import traceback
+
+import pymongo
+from ebi_eva_common_pyutils.mongodb import MongoDatabase
+
+
+def get_SHA1(variant_rec):
+    """Calculate the SHA1 digest from the seq, study, contig, start, ref, and alt attributes of the variant"""
+    h = hashlib.sha1()
+    keys = ['seq', 'study', 'contig', 'start', 'ref', 'alt']
+    h.update('_'.join([str(variant_rec[key]) for key in keys]).encode())
+    return h.hexdigest().upper()
+
+
+def replace_with_correct_contig(mongo_source):
+    correct_contig = 'AF010406.1'
+    sve_collection = mongo_source.mongo_handle[mongo_source.db_name]["submittedVariantEntity"]
+    filter_criteria = {'seq': 'GCA_000298735.1', 'study': 'PRJEB33693', 'contig': '-'}
+    cursor = sve_collection.find(filter_criteria, no_cursor_timeout=True)
+    insert_statements = []
+    drop_statements = []
+    number_of_variants_to_replace = 78
+    total_inserted, total_dropped = 0, 0
+    try:
+        for variant in cursor:
+            original_id = get_SHA1(variant)
+            assert variant['_id'] == original_id, "Original id is different from the one calculated %s != %s" % (
+                variant['_id'], original_id)
+            variant['contig'] = correct_contig
+            variant['_id'] = get_SHA1(variant)
+            insert_statements.append(pymongo.InsertOne(variant))
+            drop_statements.append(pymongo.DeleteOne({'_id': original_id}))
+        result_insert = sve_collection.bulk_write(requests=insert_statements, ordered=False)
+        total_inserted += result_insert.inserted_count
+        result_drop = sve_collection.bulk_write(requests=drop_statements, ordered=False)
+        total_dropped += result_drop.deleted_count
+        logging.info('%s / %s new documents inserted' % (total_inserted, number_of_variants_to_replace))
+        logging.info('%s / %s old documents dropped' % (total_dropped, number_of_variants_to_replace))
+    except Exception as e:
+        print(traceback.format_exc())
+        raise e
+    finally:
+        cursor.close()
+    return total_inserted
+
+
+def delete_variants_wront_contig(mongo_source):
+    sve_collection = mongo_source.mongo_handle[mongo_source.db_name]["submittedVariantEntity"]
+    filter_criteria = {'seq': 'GCA_000298735.1', 'study': 'PRJEB33693', 'contig': 'OARMT'}
```

Comment:
Since it has been a [year since the accessions have been published](https://www.ebi.ac.uk/panda/jira/browse/ELOAD-531?focusedCommentId=323989&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-323989), could you or @BaronKoy check with the submitter + FTP logs to understand if the accessions have been referenced anywhere (for example, in publications) so that we can determine the correct course of action ([merging](https://github.com/EBIvariation/eva-tasks/blob/5411bd9a1187140480efd04fa159bd0b4aad0123/tasks/eva_2146/merge_prjeb25004_redundant_ssids.groovy) might be preferable if they have already been used) for redundant variants.

Reply:
Discussed offline and @andresfsilva confirmed that the OARMT accessions in PROD are not being used in the accessioning file. So the deletion can go ahead as planned.